src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -74,7 +76,7 @@ internal partial class Binder
 
                 // We need to preserve any conversion that changes the type (even identity conversions, like object->dynamic),
                 // or that was explicitly written in code (so that GetSemanticInfo can find the syntax in the bound tree).
-                if (!isCast && source.Type.Equals(destination, TypeCompareKind.IgnoreNullableModifiersForReferenceTypes))
+                if (!isCast && source.Type!.Equals(destination, TypeCompareKind.IgnoreNullableModifiersForReferenceTypes))
```

Comment:
>Type [](start = 38, length = 4)

how do we know `source.Type` isn't null from surrounding code?

Reply:
> Note: a few lines below we do a null-test on source.Type

`BindToNaturalType` _should_ take care of that, but I'll add an assert.

---
In reply to: [367183378](https://github.com/dotnet/roslyn/pull/40946#discussion_r367183378) [](ancestors = 367183378)